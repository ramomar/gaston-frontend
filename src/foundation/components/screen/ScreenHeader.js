import React from 'react';
import PropTypes from 'prop-types';
import { Box } from 'grommet';

function ScreenHeader(props) {
  return (
    <Box
      flex={false}
      direction='row'
      align='center'
      justify='between'
      margin={{ vertical: 'large' }}
      pad={{ horizontal: 'medium' }}>
      <Box>
        {props.start}
      </Box>
      <Box>
        {props.center}
      </Box>
      <Box>
        {props.end}
      </Box>
    </Box>
  );
}

ScreenHeader.propTypes = {
  start: PropTypes.element,
  center: PropTypes.element,
  end: PropTypes.element
};

export default ScreenHeader;
