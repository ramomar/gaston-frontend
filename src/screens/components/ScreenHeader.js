import React from 'react';
import { Box, Text } from 'grommet';

function ScreenHeader({ start, center, end }) {
  return (
    <Box
      direction='row'
      height='xsmall'
      align='center'
      justify='between'
      pad={{ horizontal: 'medium' }}>
      <Box>
        {start}
      </Box>
      <Box>
        {center}
      </Box>
      <Box>
      </Box>
    </Box >
  );
}

export default ScreenHeader;
