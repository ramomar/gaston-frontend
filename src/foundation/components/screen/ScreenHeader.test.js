import React from 'react';
import renderer from 'react-test-renderer';
import ScreenHeader from './ScreenHeader';

it('renders correctly', () => {
  const component =
    <ScreenHeader
      start={<div>Start</div>}
      center={<div>Center</div>}
      end={<div>End</div>}>
    </ScreenHeader>;

  const tree = renderer.create(component);

  expect(tree).toMatchSnapshot();
});
