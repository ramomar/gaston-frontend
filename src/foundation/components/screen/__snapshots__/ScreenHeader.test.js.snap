// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 fFWPAz"
>
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
  >
    <div>
      Start
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
  >
    <div>
      Center
    </div>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 fWrrBh"
  >
    <div>
      End
    </div>
  </div>
</div>
`;
